torchquantum/layer/layers/__init__.py

```diff
@@ -0,0 +1,47 @@
+from .module_from_ops import QuantumModuleFromOps
+from .op_all import TrainableOpAll, ClassicalInOpAll, FixedOpAll, TwoQAll
+from .random_layers import RandomLayer, RandomLayerAllTypes, RandomOp1All
+from .swap_layer import SWAPSWAPLayer, SWAPSWAPLayer0
+from .cx_layer import CXLayer, CXCXCXLayer
+from .ry_layer import (
+    RYRYCXLayer0,
+    RYRYRYCXCXCXLayer0,
+    RYRYRYLayer0,
+    RYRYRYSWAPSWAPLayer0,
+)
+from .u3_layer import U3CU3Layer0, CU3Layer0
+from .qft_layer import QFTLayer
+from .seth_layer import SethLayer0, SethLayer1, SethLayer2
+from .layers import (
+    SimpleQLayer,
+    Op1QAllLayer,
+    LayerTemplate0,
+    CXRZSXLayer0,
+    RZZLayer0,
+    BarrenLayer0,
+    FarhiLayer0,
+    MaxwellLayer0,
+    RXYZCXLayer0,
+)
+
+# layer (children of LayerTemplate0) to add to the layer_name_dict
+_all_layers = [
+    SWAPSWAPLayer0,
+    RYRYCXLayer0,
+    RYRYRYCXCXCXLayer0,
+    RYRYRYLayer0,
+    RYRYRYSWAPSWAPLayer0,
+    CXRZSXLayer0,
+    RZZLayer0,
+    BarrenLayer0,
+    FarhiLayer0,
+    MaxwellLayer0,
+    RXYZCXLayer0,
+]
+
+layer_name_dict = {}
+
+for _lyr in _all_layers:
+    # check the layer has a non-empty name
+    assert _lyr.name is not None, f"Layer name not defined for {layer}"
+    layer_name_dict[_lyr.name] = _lyr
```

Comment:
Dynamically created the layer list here. Note that instead of specifying the dunder all (and therefore needing to rewrite everything again), I just used underscores to indicate not to export the variables.

Another thing to note is that I noticed the previous layer dictionary only had LayerTemplate0s (which is what’s included in the _all_layers). This dictionary, however, is only used in QuantumNode. Perhaps this can be reformatted for different purposes going forward.

Reply:
Also, for later reference, some discourse on using underscore versus __all__: https://stackoverflow.com/questions/44834/what-does-all-mean-in-python

Reply:
Another quick note: it seems like the underscores are more popular in libraries (qiskit, pennylane) than using the dunder all.